### PR TITLE
`createStore` hook & Strong typed store values

### PR DESCRIPTION
**Pull request by omarluq**
this PR:
* introduces a factory function `createStore` that simplifies the process of creating new instances of the Store class. It takes an options object as a parameter, which includes the name, initialValue, and type for the new store. This function ensures that the store is properly initialized and that the name is initially a string and converts it to a symbol for uniqueness.
* updates Store constructor to accept a type argument and introduces strong typing to the Store class. The Store class now uses a generic T to represent the type of the value it holds. This ensures type safety when getting and setting the store's value. also added type checking in the set method to ensure that the new value matches the expected type.
* updates `index.ts` to export the createStore function instead of the Store class. This change encourages the use of the createStore function to create new stores, which ensures that they are properly initialized and named. This makes the Store class more of an implementation detail, which can help to simplify the API and make the library easier to use.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,9 +47,9 @@ Let's dive into a simple use case to see how Stimulus Store works. In this examp
 
 ```js
 // controllers/stores/counter.js
-import { Store } from "stimulus-store"; // Import the shared Store class
+import { createStore } from "stimulus-store";
 
-export const counterStore = new Store(0); // Initialize with an initial value of 0
+export const counterStore = createStore(name: 'counterStore', initialValue: 0, type: Number)
```
Comment (tcarac) [inline README.md:52]: Question about this:
- Why can't it be achieved with generics + inference  vs recreating a type system?

Comment (omarluq) [reply README.md]: To enforce type safety at runtime when the store sets a value

https://github.com/omarluq/stimulus-store/blob/ae8b4cd75bcdf235c0aa232d3f3d1ad5706e480c/src/store.ts#L51-L60

And here using generics and interfaces would have made the API more complex and harder to use. I feel like this simple typing gives us all we need without overweight.
https://github.com/omarluq/stimulus-store/blob/ae8b4cd75bcdf235c0aa232d3f3d1ad5706e480c/src/createStore.ts#L34-L64

**Comment by omarluq** [comment]
@tcarac how do you feel about the switch to initializing the stores in a single file and async manner?
```js
// controllers/stores.js
import { createStore } from "stimulus-store";

export let counterStore;

(async function initializeStores() {
  counterStore = await createStore({ name: 'counterStore', initialValue: 0, type: Number });
})();
```

**Comment by omarluq** [comment]
> > @tcarac what do you think about the switch to initializing the stores in a single file and async manner?
> > ```js
> > // controllers/stores.js
> > import { createStore } from "stimulus-store";
> > 
> > export let counterStore;
> > export let anotherStore;
> > 
> > (async function initializeStores() {
> >   counterStore = await createStore({ name: 'counterStore', initialValue: 0, type: Number });
> >   anotherStore = await createStore({ name: 'anotherStore', initialValue: 'anotherStore', type: String });
> > })();
> > ```
> 
> Seems useful, but I'm not a fan of IIF. If I were to create different stores in different files I would have to rather wrap them in async functions or just create JSON objects. I think I liked the sync version more

I agree Im not huge fan of using IIFE here too!
on the other hand it makes sense that `createStore` became async since `store.set` can now accept a promise or a callback. 
This can also be a module lvl await
```js
import { createStore } from "stimulus-store";

export const counterStore = await createStore({ name: 'counterStore', initialValue: 0, type: Number });
```

or 
```js
import { createStore } from "stimulus-store";

try {
  export const counterStore = await createStore({ name: 'counterStore', initialValue: 0, type: Number });
} catch (error) {
  console.log(`Failed to create store: ${error.message}`)
}
```
this completely legal syntax! the only "gotcha" here is that the store module will not load if the promise is not resolved

**Pull request merged**